server/tests/models/test_causal_lm.py

```diff
@@ -54,38 +53,43 @@ def default_multi_requests_causal_lm_batch(default_pb_request, gpt2_tokenizer):
 
     batch_pb = generate_pb2.Batch(id=1, requests=[req_0, req_1], size=2)
     return CausalLMBatch.from_pb(
-        batch_pb, gpt2_tokenizer, torch.float32, torch.device("cpu")
+        batch_pb, gpt2_tokenizer, torch.float32, torch.device("hpu")
     )
 
 
 def test_batch_from_pb(default_pb_batch, default_causal_lm_batch):
     batch = default_causal_lm_batch
 
     assert batch.batch_id == default_pb_batch.id
-    assert batch.requests == default_pb_batch.requests
+    assert len(batch.requests) == len(default_pb_batch.requests)
+
+    for r in range(0,len(default_pb_batch.requests)):
+        assert batch.requests[r].data == default_pb_batch.requests[r]
+
+    # For Gaudi we are adding padding of multiplication of bucket size
+    size_of_padded_to_bucket = ((default_pb_batch.size + PREFILL_BATCH_BUCKET_SIZE - 1) // PREFILL_BATCH_BUCKET_SIZE) * PREFILL_BATCH_BUCKET_SIZE
```

Comment:
It looks good, but if you want to simplify you can also import round_up function from [causal_lm.py](https://github.com/huggingface/tgi-gaudi/blob/habana-main/server/text_generation_server/models/causal_lm.py#L59)

```
from text_generation_server.models.causal_lm import round_up
...
size_of_padded_to_bucket = round_up(default_pb_batch.size, PREFILL_BATCH_BUCKET_SIZE)
```

Reply:
@kdamaszk It is a valid suggestion , but I have just checked that there are not tests for this round_up function, so just to prepare for possible situation that someone will break round_up() in a code, I think I would like to leave it that way